keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ResourceKind.kt

```diff
@@ -0,0 +1,25 @@
+package com.netflix.spinnaker.keel.api
+
+data class ResourceKind(
+  val group: String,
+  val kind: String,
+  val version: String
+) {
+  init {
+    require(version.matches(Regex("\\d+(\\.\\d+)*"))) {
+      "$version is not a valid version number"
+    }
+  }
+
+  override fun toString(): String {
+    return "$group/$kind@v$version"
+  }
+
+  companion object {
+    @JvmStatic
+    fun parseKind(value: String) =
+      Regex("([\\w.-]+)/([\\w.-]+)@v(.+)").matchEntire(value)?.let { matchResult ->
+        ResourceKind(matchResult.groupValues[1], matchResult.groupValues[2], matchResult.groupValues[3])
+      } ?: error("$value is not a valid resource kind")
+  }
+}
```

Comment:
Okay, here's an actual type that represents the `kind` of the resource. The format is enforced by this, but it serializes/deserializes as a string.

This means we don't accidentally create tests with invalid kinds, among other things.